server/services/stadiums.ts

```diff
@@ -109,3 +110,145 @@ export const getStadiumProfileService = async (
+
+  const isStadiumExist = await Stadium.findOne({
+    where: { UserId: StadiumId },
+  });
+  if (!isStadiumExist) {
+    return {
+      status: 401,
+      data: 'هذا الملعب غير متاح',
+    };
+  }
+  await updatedValueSchema.validateAsync(body);
+
+  const stadiumResult = await Stadium.update(body, {
+    where: { UserId: StadiumId },
+  });
+
+  const UserResult = await User.update(body, { where: { id: StadiumId } });
+
+  return {
+    status: 200,
+    data: {
+      message: 'تم الحفظ بنجاح',
+      stadiumResult: [stadiumResult as Array<number>][0][0],
+      UserResult: [UserResult as Array<number>][0][0],
+    },
+  };
+};
+
+export const UpdateStadiumGalleryService = async (
+  req: Request,
+): Promise<{
+  status: number;
+  data: string | object;
+}> => {
+  const {
+    body: { image, id, StadiumId },
+  } = req;
+
+  // const { StadiumId } = req.UserData this will happen after create protected routes
+  const checkAuthId = 5; //and this will removed
+
+  if (StadiumId !== checkAuthId) {
+    return {
+      status: 401,
+      data: 'UnAuthorize',
+    };
+  }
+
+  const check = await Gallery.findAll({ where: { id, StadiumId } });
+  if (!check.length) {
+    return {
+      status: 401,
+      data: ' هذه الصورة غير موجودة',
+    };
+  }
+
+  const galleries = await Gallery.update(
+    { image },
+    { where: { id, StadiumId }, returning: true },
+  );
+
+  return {
+    status: 200,
+    data: galleries,
+  };
+};
+
+export const AddStadiumImageService = async (
+  req: Request,
+): Promise<{
+  status: number;
+  data: string | object;
+}> => {
+  const { body } = req;
+  const { StadiumId } = body;
+
+  // const { StadiumId } = req.UserData this will happen after create protected routes
+  const checkAuthId = 5; //and this will removed
+
+  if (StadiumId !== checkAuthId) {
+    return {
+      status: 401,
+      data: 'UnAuthorize',
+    };
+  }
+
+  const check = await Gallery.findAll({ where: { StadiumId } });
+  if (check.length >= 4) {
+    return {
+      status: 401,
+      data: 'لا يمكن اضافة اكثر من اربعة صور',
+    };
+  }
+
+  const galleries = await Gallery.create({ ...body, StadiumId });
+
+  return {
+    status: 200,
+    data: galleries,
+  };
+};
+
+export const deleteStadiumImageService = async (
+  req: Request,
+): Promise<{ status: number; data: string | number }> => {
+  const { ImageId, StadiumId } = req.params;
+  // const checkAuthId = req.UserData.StadiumId; this will happen after create protected routes
+  const checkAuthId = 5;
+
+  if (+StadiumId !== checkAuthId) {
+    return {
+      status: 401,
+      data: 'UnAuthorize',
+    };
+  }
+
+  const check = await Gallery.findOne({
+    where: { StadiumId: +StadiumId, id: +ImageId },
+  });
+  if (!check) {
+    return {
+      status: 401,
+      data: 'هذه الصورة غير موجودة',
+    };
+  }
+
+  const result = await Gallery.destroy({
+    where: { StadiumId: +StadiumId, id: +ImageId },
+  });
+
+  return { status: 204, data: result };
```

Comment:
`result` here is the number of deleted rows, what the benefit from it to send it in the response